domain_models/models.py

```diff
@@ -181,5 +182,36 @@ def __str__(self):
     @property
     def __data__(self):
         """Read only dictionary of model fields/values."""
-        return dict((field.name, field.get_value(self))
-                    for field in self.__class__.__fields__)
+        return dict((name, field.get_value(self))
+                    for name, field in
+                    six.iteritems(self.__class__.__fields__))
+
+    def get(self, field_name, default=None):
+        """Return the value of the field.
+
+        Analogue for `dict.get()` python method.
+        `field_name` must be a string. If the string is the name of
+        one of the existent fields, the result is the value of that field.
+        For example, `model.get('foobar')` is equivalent to `model.foobar`.
+        If the filed does not have a value, `default` is returned if provided.
+        It will raise `AttributeError` if `default` can not be converted to
+        right type value.
+        If the field does not exist, `AttributeError` is raised as well.
+
+        :param string field_name:
+        :param mixed default:
+        """
+        if not hasattr(self, field_name):
```

Comment:
I got one idea about the implementation of this thing... Here is it:

```python
try:
    field = self.__class__.__fields__[field_name]
except KeyError:
    raise AttributeError
else:
    return field.get_value(self, default)
```

If so, we will move `default` converting logic closer to the converter. Also scope of field converters could be turned back to protected. It can have some sense...